src/meta/src/hummock/manager/mod.rs

```diff
@@ -1166,10 +1172,82 @@ where
                 ..Default::default()
             },
         );
+
         let mut new_hummock_version = old_version;
-        new_hummock_version.id = new_version_id;
         new_version_delta.id = new_version_id;
+        new_hummock_version.id = new_version_id;
 
+        for group_id in old_version_groups {
```

Comment:
It would be better to split these codes to another function and have some comments on it to make it easy to read.

Reply:
> It would be better to split these codes to another function and have some comments on it to make it easy to read.

ok. let me refactor.